batch/src/main/java/org/openmrs/analytics/FetchPatientHistory.java

```diff
@@ -13,25 +13,22 @@
 // limitations under the License.
 package org.openmrs.analytics;
 
+import java.io.IOException;
 import java.util.List;
 
 import ca.uhn.fhir.rest.api.SummaryEnum;
 import com.google.common.base.Preconditions;
-import com.google.common.collect.Lists;
 import org.apache.avro.Schema;
-import org.apache.avro.generic.GenericRecord;
-import org.apache.beam.sdk.coders.AvroCoder;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.transforms.ParDo;
 import org.apache.beam.sdk.values.KV;
 import org.apache.beam.sdk.values.PCollection;
-import org.apache.beam.sdk.values.PCollectionTuple;
-import org.apache.beam.sdk.values.TupleTagList;
+import org.apache.beam.sdk.values.PDone;
 import org.hl7.fhir.r4.model.Bundle;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class FetchPatientHistory extends PTransform<PCollection<KV<String, Integer>>, PCollectionTuple> {
+public class FetchPatientHistory extends PTransform<PCollection<KV<String, Integer>>, PDone> {
```

Comment:
What does PDone do?

Reply:
[PDone](https://beam.apache.org/releases/javadoc/2.20.0/org/apache/beam/sdk/values/PDone.html) is used for transforms that are terminal, i.e., they have no output. I think they are usually used for output transforms.